apps/web/src/app/_components/form/review-form.tsx

```diff
@@ -262,20 +264,46 @@ export function ReviewForm(props: ReviewFormProps) {
     scroll.scrollToTop({ duration: 250, smooth: true });
   };
 
-  const mutation = api.review.create.useMutation();
+  const [validForm, setValidForm] = useState(true); //wait where do i put these
+  const { toast } = useToast();                     //like at the tipity top?
 
-  function onSubmit(values: z.infer<ReviewFormType>) {
-    mutation.mutate({
-      roleId: props.roleId,
-      profileId: props.profileId,
-      companyId: props.company.id,
-      ...values,
-    });
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+
+  const mutation = api.review.create.useMutation({
+    onError: (error) => {
+      setValidForm(false);
+      setErrorMessage(error.message || "An unknown error occurred.");
+  
+      toast({
+        title: "Submission Error",
+        description: (error && error.message) ? error.message : "Something went wrong. Please try again.",
+        variant: "destructive",
+      });
+    },
+  });
+
+
+  
+
+  async function onSubmit(values: z.infer<ReviewFormType>) {
+    try {
+      await mutation.mutateAsync({
+        roleId: props.roleId,
+        profileId: props.profileId,
+        companyId: props.company?.id ?? "",
```

Comment:
I think this can just be `props.company.id` since company is always defined in the props and id is a non-null value of company